Book/php7/extensions_design/extension_skeleton.rst

```diff
@@ -116,7 +116,7 @@ The ``ini_entry`` vector is actually unused. You :doc:`register INI entries <ini
 Then you may declare dependencies, that means that your extension could need another extension to be loaded before it,
 or could declare a conflict with another extensions. This is done using the ``deps`` field. In reality, this is very
 ucommonly used, and more generally it is a bad practice to create dependencies across PHP extensions. Also, note that
-dependencies are
+dependencies are ``RTLD_NOW`` on most distros, and thus pretty useless.
```

Comment:
`ucommonly` in the line above, fix that too? In any case, I would write this as:

note that dependencies are ``RTLD_NOW`` on most Linux distributions. Defining dependencies is mostly useful to make certain that initialisation happens in the right order, so that an extension can depend on another extension's initialisation to already have occurred.

Reply:
I do also disagree that this is bad practise → You know from your own code base that it is necessary to depend on the MongoDB extension, so that its initialisation is done before you use it.

Reply:
`RTLD_NOW` makes shared object resolve all their symbols upon loading, and not just when they are being used. This makes sure that all the symbols (functions) that an extension requires definitely exist, and won't throw *runtime* warnings when unavailable symbols are attempted of being used.